analytics-api-core/src/main/scala/org/ekstep/analytics/api/util/PostgresDBUtil.scala

```diff
@@ -1,97 +1,158 @@
-  }
-
-  def readGeoLocationRange(sqlString: String): List[GeoLocationRange] = {
-    SQL(sqlString).map(rs => GeoLocationRange(rs)).list().apply()
-  }
-
-  def checkConnection = {
-    try {
-      val conn = ConnectionPool.borrow()
-      conn.close()
-      true
-    } catch {
-      case ex: Exception => 
-        ex.printStackTrace();
-        false
+    private lazy val db = AppConfig.getString("postgres.db")
+    private lazy val url = AppConfig.getString("postgres.url")
+    private lazy val user = AppConfig.getString("postgres.user")
+    private lazy val pass = AppConfig.getString("postgres.pass")
+
+    Class.forName("org.postgresql.Driver")
+    ConnectionPool.singleton(s"$url$db", user, pass)
+
+    implicit val session: AutoSession = AutoSession
+
+    def read(sqlString: String): List[ConsumerChannel] = {
+        SQL(sqlString).map(rs => ConsumerChannel(rs)).list().apply()
+    }
+
+    def readLocation(sqlString: String): List[DeviceLocation] = {
+        SQL(sqlString).map(rs => DeviceLocation(rs)).list().apply()
+    }
+
+    def readGeoLocationCity(sqlString: String): List[GeoLocationCity] = {
+        SQL(sqlString).map(rs => GeoLocationCity(rs)).list().apply()
+    }
+
+    def readGeoLocationRange(sqlString: String): List[GeoLocationRange] = {
+        SQL(sqlString).map(rs => GeoLocationRange(rs)).list().apply()
+    }
+
+
+    def saveReportConfig(reportRequest: ReportRequest): String = {
+        val config = JSONUtils.serialize(reportRequest.config)
+        sql"""insert into ${ReportConfig.table}(report_id, updated_on, report_description, requested_by,
+             report_schedule, config, created_on, submitted_on, status, status_msg) values
+              (${reportRequest.reportId}, ${new Date()}, ${reportRequest.description},
+              ${reportRequest.createdBy},${reportRequest.reportSchedule} , CAST($config AS JSON),
+              ${new Date()}, ${new Date()} ,'SUBMITTED', 'REPORT SUCCESSFULLY SUBMITTED')""".update().apply().toString
+    }
+
+
+    def updateReportConfig(reportId: String, reportRequest: ReportRequest): String = {
+        val config = JSONUtils.serialize(reportRequest.config)
+        val q =
+            sql"""update ${ReportConfig.table} set updated_on =${new Date()} ,
+             report_description = ${reportRequest.description}, requested_by = ${reportRequest.createdBy} ,
+             report_schedule = ${reportRequest.reportSchedule} , config = ($config::JSON) ,
+               status = 'SUBMITTED' , status_msg = 'REPORT SUCCESSFULLY SUBMITTED'  where report_id =$reportId"""
+        q.update().apply().toString
+    }
+
+    def readReport(reportId: String): Option[ReportConfig] = {
+        sql"select * from ${ReportConfig.table} where report_id = ${reportId}".map(rc => ReportConfig(rc)).first().apply()
+    }
+
+    def deleteReport(reportId: String) = {
+        sql"delete from ${ReportConfig.table} where report_id=$reportId".execute().apply()
+
```

Comment:
We are not deleting report configs. We are just deactivating them.

Reply:
Done